applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -962,7 +975,13 @@ static void report_issued(const void *subscriber_id, uint8_t report_id, bool err
 		if ((next_rs->state != STATE_DISCONNECTED) &&
 		    (next_rs->linked_rd->linked_rs == next_rs) &&
 		    (next_rs->cnt == 0)) {
-			if (report_send(next_rs->linked_rd, false, false)) {
+			if (report_send(next_rs->linked_rd, next_rs, false, false)) {
+				break;
+			}
+		}
+
+		if (next_rs->update_needed && next_rs->linked_rd == &empty_rd) {
```

Comment:
I think it would be enough to add `(next_rs->linked_rd->linked_rs == NULL)`

i.e.
```
		if ((next_rs->state != STATE_DISCONNECTED) &&
		    (next_rs->cnt == 0) &&
		    ((next_rs->linked_rd->linked_rs == next_rs) ||
		     (next_rs->linked_rd->linked_rs == NULL)) {
```

Reply:
then you don't need to add any additional call to report_send as line 978 will be enough for all cases